### PR TITLE
createDto Mapper and controller

### PR DESCRIPTION
**Pull request by don-bigdad**

### CODE REVIEW

File: src/main/java/com/example/springbootbookshop/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping(value = "/{id}")
```
Comment (teract10s) [inline src/main/java/com/example/springbootbookshop/controller/BookController.java:27]: you can left only @GetMapping("/{id}") without param value

File: src/main/java/com/example/springbootbookshop/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (teract10s) [inline src/main/java/com/example/springbootbookshop/controller/BookController.java:18]: here the same

File: src/main/java/com/example/springbootbookshop/mapper/BookMapper.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import com.example.springbootbookshop.dto.BookDto;
+import com.example.springbootbookshop.dto.CreateBookRequestDto;
+import com.example.springbootbookshop.entity.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book bookToModel(CreateBookRequestDto bookDto);
```
Comment (teract10s) [inline src/main/java/com/example/springbootbookshop/mapper/BookMapper.java]: Maybe rename it and left just "toModel"

**Review by teract10s — APPROVED** [review]
Good job

File: checkstyle.xml [hunk 4]
```diff
@@ -84,6 +84,7 @@
             <message key="ws.notPreceded"
                      value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
         </module>
+
```
Comment (NazarSmith228) [inline checkstyle.xml]: try not to modify existing code when possible :)

File: src/main/java/com/example/springbootbookshop/config/MapperConfig.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
+public class MapperConfig {
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/config/MapperConfig.java]: it is better to specify `@MapperConfig` directly above the mapper class

File: src/main/java/com/example/springbootbookshop/dto/BookDto.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.example.springbootbookshop.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/dto/BookDto.java]: it is better to use records for DTO

if your Checkstyle fires at it - upgrade `maven-checkstyle-plugin` to 3.3.0

File: src/main/java/com/example/springbootbookshop/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.example.springbootbookshop.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/dto/CreateBookRequestDto.java]: it is better to use records for DTO

if your Checkstyle fires at it - upgrade `maven-checkstyle-plugin` to 3.3.0

File: src/main/java/com/example/springbootbookshop/mapper/BookMapper.java (hunk 3 above)
Comment (NazarSmith228) [reply src/main/java/com/example/springbootbookshop/mapper/BookMapper.java]: I would recommend `toBook`

File: src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java [hunk 8]
```diff
@@ -50,4 +51,16 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can`t get all books from DB", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.createQuery("from Book where id = :id", Book.class)
+                    .setParameter("id",id)
+                    .uniqueResult());
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java]: you can replace it with a simpler approach:
```
Optional.ofNullable(sessionFactory.fromSession(s -> s.find(Book.class, id)));
```

File: src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java [hunk 9]
```diff
@@ -50,4 +51,16 @@ public List<Book> findAll() {
...
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t get Book from DB with id: "
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/repository/impl/BookRepositoryImpl.java]: I believe you should throw `EntityNotFoundException` here

File: src/main/java/com/example/springbootbookshop/service/impl/BookServiceImpl.java [hunk 10]
```diff
@@ -1,28 +1,38 @@
...
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto getBookById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "Can`t get book with id from DB: " + id)));
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/service/impl/BookServiceImpl.java]: we can make use of Optional's `map` method here:
```
return repository.findById(id)
            .map(bookMapper::toDto)
            .orElseThrow(...);
```

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**